jvm-runtime/plugin/common/jvmcommon.go

```diff
@@ -167,6 +172,9 @@ func (s *Service) SyncStubReferences(ctx context.Context, req *connect.Request[l
 // rebuild must include the latest build context id provided by the request or subsequent BuildContextUpdated
 // calls.
 func (s *Service) Build(ctx context.Context, req *connect.Request[langpb.BuildRequest], stream *connect.ServerStream[langpb.BuildEvent]) error {
+	if req.Msg.RebuildAutomatically {
+		return s.handleDevModeRequest(ctx, req, stream)
+	}
 	events := make(chan updateEvent, 32)
```

Comment:
the rest of this function can be gutted I think as it has a lot of logic for receiving new build contexts and watching for file changes, which only happens if `req.Msg.RebuildAutomatically` is true.

Oh wait I'm actually not sure how you are reacting to build context changes?

Reply:
I am not handling that properly I think, I need to move some of this logic into the hot reload path.

Reply:
I think I am handling this properly now